File: src/ios/Integrity.m
#import "Integrity.h"
#import <Cordova/CDVPlugin.h>

@implementation Integrity

- (void)check:(CDVInvokedUrlCommand*)command {
  NSString* alias = [command.arguments objectAtIndex:0];
  NSData* tag = [alias dataUsingEncoding:NSUTF8StringEncoding];
  CDVPluginResult* pluginResult = nil;
  if (alias == nil)
    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Error"];
  else
    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Success"”];
  [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end

File: src/ios/Integrity.swift
import DeviceCheck

@objc(Integrity)
class Integrity : CDVPlugin {
  @objc func check(_ command: CDVInvokedUrlCommand) {
    if DCDevice.current.isSupported {
      DCDevice.current.generateToken(completionHandler: {
        token, error in guard let token = token else {
          let pluginResult = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAs: "Failed to create token")
          self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
          return;
        }
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: token.base64EncodedString())
        self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
        return;
      })
    } else {
      let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: "N/A")
      self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
    }
  }
}
